src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -112,6 +112,14 @@ static ucs_status_t uct_tcp_sockcm_ep_progress_send(uct_tcp_sockcm_ep_t *cep)
     if (uct_tcp_sockcm_ep_is_tx_rx_done(cep)) {
         ucs_assert(status == UCS_OK);
         cep->state |= UCT_TCP_SOCKCM_EP_DATA_SENT;
+
+        /* on the client side - if completed sending a message after the connect_cb
+         * was already invoked, then this message is the notify message */
+        if (cep->state & UCT_TCP_SOCKCM_EP_CLIENT_CONNECTED_CB_INVOKED) {
+            ucs_assert(cep->state & UCT_TCP_SOCKCM_EP_ON_CLIENT);
+            cep->state |= UCT_TCP_SOCKCM_EP_CLIENT_NOTIFY_SENT;
```

Comment:
how do we know the user already called notify() API?

Reply:
the notify message is the only message that should be sent from the client side after the client invoked its connect_cb. do we have another way of knowing that the client finished sending the notify message?

Reply:
maybe add another flag that notify was called and flag that notify was sent